src/main/java/net/dv8tion/jda/api/utils/data/DataObject.java

```diff
@@ -620,6 +646,25 @@ else if (value instanceof DataArray)
         return data.keySet();
     }
 
+    /**
+     * Serialize this object as JSON.
+     *
+     * @return a byte array containing the JSON representation of this object.
+     */
+    public byte[] toJson()
+    {
+        try
+        {
+            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+            mapper.writeValue(outputStream, data);
+            return outputStream.toByteArray();
+        }
+        catch (IOException e)
+        {
+            throw new ParsingException(e);
```

Comment:
Why did you use a different exception here?

Reply:
Just copied the `toString()` method from right below, no big thoughts went into this. I don't understand the structure and failure handling of JDA good enough to make an educated decision anyways, feel free to edit at will.